src/hexadecimal.jl

```diff
@@ -176,9 +177,12 @@ end
 @noinline function _backtrack_error(source, pos, len, code, segment_len, hi, lo, pl)
     pos -= segment_len # Backtrack to the start of the invalid hex
     fastseek!(source, pos - 1)
-    while _HEX_LUT[peekbyte(source, pos) + 0x01] != 0xFFFF
+    b = peekbyte(source, pos)
```

Comment:
should we check `pos` is valid here?

maybe a couple lines up add an `@assert 1 <= pos <= len`(is that right?)

Reply:
`pos` should always be valid here -- we only use `_backtrack_error` once we check the length would fit a UUID value and then we iterate in fixed sized steps (``segment_len`).